packages/muelu/test/scaling/Driver.cpp

```diff
@@ -494,13 +494,13 @@ int main_(Teuchos::CommandLineProcessor &clp, Xpetra::UnderlyingLib& lib, int ar
         tm = Teuchos::null;
       }
 
-      int* tempVector;
+      std::vector<int> tempVector;
       int min = 0, max = 10;
       int numInts = 0;
       if (cacheSize > 0) {
         cacheSize *= 1024; //convert to bytes
         numInts = cacheSize/sizeof(int) + 1;
-        tempVector = new int[numInts];
+        tempVector.reserve(numInts);
```

Comment:
Shouldn't this be tempVector.resize(numInts)?

Reply:
@brian-kelley `reserve` allocates the space but doesn't initialize it or change the size of the vector.  `resize` allocates the space, initializes it, and sets the size of the vector to `numInts`.  I think in this case, the two would be equivalent, as the vector is accessed via `[ ]`, not `push_back`.

Reply:
@jhux2 No. When you `reserve`, you change capacity, but the vector size is still 0. Thus, accessing any element is improper (though will still work without array bounds check). At no point in the proposed code you actually set its size to non-zero.

Reply:
@aprokop @brian-kelley Huh, you're right.  I did a little more reading -- since the vector is size 0, using `[ ]` is undefined behavior.  Thanks!